twilight-cache-inmemory/src/event/presence.rs

```diff
@@ -34,7 +34,11 @@ impl UpdateCache for PresenceUpdate {
 
         let presence = CachedPresence::from_model(self.0.clone());
 
-        cache.cache_presence(self.guild_id, presence);
+        // Presence updates should always contain a guild ID as per the API docs.
+        cache.cache_presence(
+            self.guild_id.expect("Presence update without guild id"),
+            presence,
+        );
```

Comment:
This assumption is not future proof!

Reply:
I may be missing something here, but it's explicitly stated in the docs that this field is always present: https://discord.com/developers/docs/topics/gateway-events#presence-update.  If discord were to stop sending `guild_id` for this event it would be a breaking change as they're changing parameters they state they send.

Reply:
The problem is that it can now deserialize (failing to do so would return an error to the user) and the later panic instead of failing to deserialize. We don't ever want to panic.

However, we can probably just conditionally cache presences when they contain a guild_id (better to potentially not cache them than to panic).

I'll have to review the full PR to give more context, but this stood out to me immediately.

Reply:
Ok I understand what you're saying now. I changed to only cache if `guild_id` is `Some`.